devices/YarpRobotLoggerDevice/src/YarpRobotLoggerDevice.cpp

```diff
@@ -1366,116 +1522,154 @@ void YarpRobotLoggerDevice::run()
     {
         std::lock_guard<std::mutex> lock(signal.mutex);
-        const BipedalLocomotion::YarpUtilities::VectorsCollection* collection
+
+        BipedalLocomotion::YarpUtilities::VectorsCollection* externalSignalCollection
             = signal.client.readData(false);
-        if (collection != nullptr)
+
+        if (externalSignalCollection != nullptr)
         {
-            if (!signal.dataArrived)
+            if(!signal.dataArrived)
             {
-                for (const auto& [key, vector] : collection->vectors)
+                for (const auto& [key, vector] : externalSignalCollection->vectors)
                 {
-                    signalFullName = signal.signalName + "::" + key;
+                    signalName = signal.signalName + TREE_DELIM + key;
                     const auto& metadata = signal.metadata.vectors.find(key);
                     if (metadata == signal.metadata.vectors.cend())
                     {
                         log()->warn("{} Unable to find the metadata for the signal named {}. The "
                                     "default one will be used.",
                                     logPrefix,
-                                    signalFullName);
-                        m_bufferManager.addChannel({signalFullName, {vector.size(), 1}});
+                                    signalName);
+                        (*this.*initMetadataFunction)(signalName, {});
                     } else
                     {
-                        // if the metadata is found we use it
-                        m_bufferManager.addChannel({signalFullName, //
-                                                    {vector.size(), 1},
-                                                    metadata->second});
+                        (*this.*initMetadataFunction)(signalName, {metadata->second});
                     }
-
-                    signal.dataArrived = true;
                 }
+                signal.dataArrived = true;
+                signal.numMissedPackets = 0;
+                newMetadata = true;
             }
+            else
+            {
+                signal.numMissedPackets = 0;
+                for (const auto& [key, vector] : externalSignalCollection->vectors)
+                {
+                    signalName = signal.signalName + TREE_DELIM + key;
+                    m_bufferManager.push_back(vector, time, signalName);
+                    if(sendDataRT)
+                    {
+                        rtSignalName
+                            = ROBOT_RT_ROOT_NAME + TREE_DELIM + signalName;
+                        m_vectorCollectionRTDataServer.populateData(rtSignalName, vector);
+                    }
+                }
 
-            for (const auto& [key, vector] : collection->vectors)
+            }
+        }
+        else if (externalSignalCollection == nullptr && signal.connected)
+        {
+            signal.numMissedPackets++;
```

Comment:
This statement is inaccurate. The logger does not consider the period of the exogenous signal. In fact, the signal may arrive at a frequency different from that of the logger, which is acceptable. I will completely remove this logic as it would introduce misleading information. For example, if the exogenous signal arrives from a controller running at 100 Hz while the logger operates at 1 kHz, there are no missing packets.

Reply:
Better to discuss face-to-face. I understand the issue here. This logic was initially put in to disconnect exogenous signals which were no longer running.